visitor/TreeVisitor.cpp

```diff
@@ -56,22 +56,22 @@ bool TreeVisitor::VisitEnumDecl(clang::EnumDecl *enumdecl){
     //Replace "enum x" with enum_x in scala
     typeTranslator.AddTranslation("enum " + name, "enum_" + name);
 
-    if(name == "" && enumdecl->getTypedefNameForAnonDecl()){
+    if(name.empty() && enumdecl->getTypedefNameForAnonDecl()){
         name = enumdecl->getTypedefNameForAnonDecl()->getNameAsString();
     }
 
-    if(name != ""){
-     declarations += "\ttype enum_" + name + " = native.CInt\n";
+    if(!name.empty()){
+        ir->getLibObject().addType("type enum_" + name + " = native.CInt");
```

Comment:
I always thought that C enums were backed by unsigned ints, but after searching a bit it sounds like it can be int or char as well.

http://en.cppreference.com/w/c/language/enum
> Each enumerated type is compatible with one of: char, a signed integer type, or an unsigned integer type. It is implementation-defined which type is compatible with any given enumerated type, but whatever it is, it must be capable of representing all enumerator values of that enumeration.

I suggest we make a ticket and revisit it later. WDYT?

Reply:
@jonas, probably this will give us the type `enumdecl->getIntegerType()`, but anyway I created a separate ticket for it #12